api/routes/link-routes.ts

```diff
@@ -68,15 +66,15 @@ export const fetchLink = async (req: Request, res: Response) => {
       const html = await generateForbidden();
       return res.status(403).send(html);
     }
-    incrementClick(req.params.shortCode, req.clientIp);
+    await incrementClick(req.params.shortCode, req.clientIp);
     const html = await generateRedirect(result);
     return res.status(200).send(html);
   }
 
   try {
     result = await fetchLinkController(req.params.shortCode);
     const html = await generateRedirect(result);
-    incrementClick(req.params.shortCode, req.clientIp);
+    await incrementClick(req.params.shortCode, req.clientIp);
```

Comment:
Await will make this piece of code to wait till the promise resolves. While ideally we should do this, in this case it isn't recommended. incrementClick only increments the count, which doesn't affect redirection. Our aim is to redirect as fast as possible and this might make that a few ms slower. Also, if there is an error in this function, we do not want that to affect redirection.

Reply:
hah makes sense, I'll revert these changes.